File: StackUsingLinkedlist.java
class StackUsingLinkedlist {
	Node top= null;
	
	public class Node {

		int data; 
		Node next; 
	}
	
	
	public void push(int x) 
	{
		Node temp = new Node();
		temp.data = x;
		temp.next = top;
		top = temp;
	}

	
	public boolean isEmpty()
	{
		return top == null;
	}

	
	public int peek()
	{
		if (!isEmpty()) {
			return top.data;
		}
		else {
			System.out.println("Stack is empty");
			return -1;
		}
	}

	
	public void pop() 
	{
		
		if (top == null) {
			System.out.print("\nStack Underflow");
			return;
		}

		top = (top).next;
	}

	public void display()
	{
		
		if (top == null) {
			System.out.printf("\nStack Underflow");
			
		}
		else {
			Node temp = top;
			while (temp != null) {
				System.out.printf("%d->", temp.data);
				temp = temp.next;
			}
		}
	}


	public static void main(String[] args)
	{
		
		StackUsingLinkedlist obj = new StackUsingLinkedlist();
		obj.push(11);
		obj.push(22);
		obj.push(33);
		obj.push(44);
		obj.display();
		System.out.printf("\nTop element is %d\n", obj.peek());
		obj.pop();
		obj.pop();
		System.out.printf("After removing two nodes\n");
		obj.display();
		System.out.printf("\nTop element is %d\n", obj.peek());
	}
}

File: Stack_Using_LinkedList.js
class Node {
constructor(data, next = null){
    this.data = data;
    this.next = next;
}
}

class LinkedList{
    constructor(){
        this.head = null;
        this.size = 0;
    }
    push(data){
        let node = new Node(data);
        let current;
        current = this.head;
        node.next = current;
        this.head = node;
        this.size++;
    }
    pop(){
        let current = this.head;
        if(current){
            let data = current.data;
            current = current.next;
            this.head = current;
            this.length--;
            return data;
        }
        return null;
    }
    peek(){
        let current = this.head;
        if(current){
            return current.data;
        }
        else{
            return null;
        }
    }
    isEmpty(){
        if(this.size > 0){
            return false;
        }
        else{
            return true;
        }
    }
}
let Stack = new LinkedList();
Stack.push(10);
Stack.push(20);
Stack.push(30);
console.log(Stack.pop() + " Popped from Stack");
console.log("Top element is " + Stack.peek());